dhnx/optimization/precalc_hydraulic.py

```diff
@@ -25,48 +25,345 @@
 #   - https://www.schweizer-fn.de/stroemung/rauhigkeit/rauhigkeit.php
```

Comment:
Move the links to the module docstring. *The calculations are based on widely used empirical formulas, see e.g.: ...*